kerl

```diff
@@ -754,7 +668,131 @@ _flags() {
     esac
 }
 
+_dpkg() {
+    # gratefully stolen from
+    # https://superuser.com/questions/427318/test-if-a-package-is-installed-in-apt
+    # returns 0 (true) if found, 1 otherwise
+    echo "dpkg-query -Wf'${db:Status-abbrev}' \"$1\" 2>/dev/null | \grep -q '^i'"
+}
+
+_rpm() {
+    echo "rpm --quiet -q \"$1\" >/dev/null 2>&1"
+}
+
+_apk() {
+    echo "apk -e info \"$1\" >/dev/null 2>&1"
+}
```

Comment:
This is basically the same as before. But we prepare the command, via an `echo` we further `eval`. Because when the probe fails I wanna show how we're probing...